tests/tests-hz38/src/main/java/com/hazelcast/simulator/tests/quorum/QuorumQueueTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.tests.quorum;
+
+import static org.junit.Assert.assertEquals;
+
+import com.hazelcast.core.IQueue;
+import com.hazelcast.quorum.QuorumException;
+import com.hazelcast.simulator.test.AbstractTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+
+public class QuorumQueueTest extends AbstractTest {
+
+	// properties
+	private IQueue<Long> queue;
+	public int keyCount = 100000;
+
+	private volatile boolean flag;
+
+
+	@Setup
+	@SuppressWarnings("unchecked")
+	public void setup() {
+		queue = targetInstance.getQueue(name);
+		try {
+			queue.put(0L);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	@TimeStep
+	public void put(BaseThreadState state) {
```

Comment:
The method name is `put()`, but you are doing a `peek()`. This could lead to some confusion if there is an error being reported.